Makefile

```diff
@@ -1,19 +1,52 @@
+# Copyright 2020 Joyent, Inc.
+
 UNAME=$(shell uname -s | tr "[:upper:]" "[:lower:]")
 VER=$(shell json -f package.json version)
+NAME := thoth
+RELEASE_TARBALL := $(NAME)-$(UNAME)-$(VER).tar.gz
+ROOT := $(shell pwd)
 
-node_modules: package.json
-	npm install --production
-	@touch node_modules
+NODE_PREBUILT_VERSION=v8.17.0
+NODE_PREBUILT_TAG=gz
+NODE_PREBUILT_IMAGE=5417ab20-3156-11ea-8b19-2b66f5e7a439
 
-.PHONY: release github-release
-release: thoth-$(UNAME)-$(VER).tar.gz
+ENGBLD_USE_BUILDIMAGE = false
+ENGBLD_SKIP_VALIDATE_BUILDENV = true
+ENGBLD_REQUIRE := $(shell git submodule update --init deps/eng)
+include ./deps/eng/tools/mk/Makefile.defs
+TOP ?= $(error Unable to access eng.git submodule Makefiles.)
+RELSTAGEDIR := /tmp/$(NAME)-$(STAMP)
 
-publish: release
-	./tools/publish.sh "$(VER)"
+include ./deps/eng/tools/mk/Makefile.node_prebuilt.defs
```

Comment:
This breaks usage of this Makefile for using thoth on non-SmartOS. sdc-imgapi.git and others work around that with somethign like:

```
ifeq ($(shell uname -s),SunOS)
	include ./deps/eng/tools/mk/Makefile.node_prebuilt.defs
else
	NPM=npm
	NODE=node
	NPM_EXEC=$(shell which npm)
	NODE_EXEC=$(shell which node)
endif
```

along with something like this in package.json to hint at the particular min node version:

```
  "engines": {
    "node": ">=8.x"
  },
```

Reply:
Thanks.